tests/app/test_base.py

```diff
@@ -1,262 +1,164 @@
-    assert str(exc.value) == ''
-    assert pre_process.call_count == 1
-    on_fail_before_post_process.assert_called_once_with(
-        event=event, state=state, **on_fail_before_post_process_kwargs
-    )
-    process.assert_not_called()
-    post_process.assert_not_called()
-
-
-def test__run_exc_in_process(base_app, patch_base_event):
-    event = 'myevent'
-    state = 'state'
-
-    on_fail_before_post_process_kwargs = {'fail_key_1': 'fail_val_1'}
-
-    pre_process_result = ProcessResult(event=BaseEvent([]))
-
-    with patch.object(base_app, 'pre_process', return_value=pre_process_result) as pre_process, \
-         patch.object(base_app, 'process', side_effect=Exception) as process, \
-         patch.object(base_app, 'on_fail_before_post_process') as on_fail_before_post_process, \
-         patch.object(base_app, 'post_process') as post_process, \
-         patch.object(base_app, 'logger') as logger_mock, \
-         pytest.raises(Exception) as exc:
-        base_app._run(
-            event=event,
-            state=state,
-            on_fail_before_post_process_kwargs=on_fail_before_post_process_kwargs
-        )
-    logger_mock.error.assert_called_once_with('An error occurred in pre_process or process.')
-    assert str(exc.value) == ''
-    assert pre_process.call_count == 1
-    assert process.call_count == 1
-    on_fail_before_post_process.assert_called_once_with(
-        event=pre_process_result.event, state=state, **on_fail_before_post_process_kwargs
-    )
-    post_process.assert_not_called()
-
-
-def test__run(base_app, patch_base_event):
-    event = 'myevent'
-    state = 'state'
-
-    pre_process_result = ProcessResult(event=BaseEvent([]))
-    process_result = ProcessResult(event=BaseEvent([]))
-
-    with patch.object(base_app, 'pre_process', return_value=pre_process_result) as pre_process, \
-         patch.object(base_app, 'process', return_value=process_result) as process, \
-         patch.object(base_app, 'on_fail_before_post_process') as on_fail_before_post_process, \
-         patch.object(base_app, 'post_process') as post_process:
-        base_app._run(event=event, state=state)
-        assert pre_process.call_count == 1
-        assert process.call_count == 1
-        assert post_process.call_count == 1
-        on_fail_before_post_process.assert_not_called()
-
-
-def test__run_except_in_post_process(base_app):
-    event = 'event'
-    state = 'state'
-
-    with patch.object(base_app, 'pre_process'), \
-         patch.object(base_app, 'process'), \
-         patch.object(base_app, 'post_process', side_effect=Exception), \
-         patch.object(base_app, 'logger') as logger_mock, \
-         pytest.raises(Exception):
-        base_app._run(event=event, state=state)
-    logger_mock.error.assert_called_once_with('An error occurred in post_process.')
+def test__run_exc_in_process(mocker: MockerFixture, base_app):
```

Comment:
why there are two underscores

Reply:
also, what this test is testing. can you please add a description or change a name, so it would be more clear?

Reply:
Decided to go with two, because the method tested is called `_run` and there is also a method `run` in same class